chef/cookbooks/crowbar/recipes/prepare-upgrade-scripts.rb

```diff
@@ -112,6 +112,20 @@
   only_if { cinder_controller && (!use_ha || is_cluster_founder) }
 end
 
+all_nodes = search(:node, "state:crowbar_upgrade").map { |n| n["hostname"] }
```

Comment:
It might be a good idea to do 2 searches, one for controller nodes and one for compute